File: tools-starter-swagger/src/main/java/cn/gjing/tools/swagger/RequestHeader.scala
package cn.gjing.tools.swagger

import scala.beans.BeanProperty

/**
 * @author Gjing
 **/
class RequestHeader {
  /**
   * 请求头名称
   */
  @BeanProperty
  var name: String = _
  /**
   * 描述
   */
  @BeanProperty
  var description: String = _
  /**
   * 是否必须, 默认true
   */
  @BeanProperty
  var required: Boolean = true

  def this(name: String, description: String, required: Boolean) {
    this()
    this.name = name
    this.description = description
    this.required = required
  }
}

File: tools-starter-swagger/src/main/java/cn/gjing/tools/swagger/RequestHeader.java
package cn.gjing.tools.swagger;

import lombok.*;

/**
 * Global request header
 *
 * @author Gjing
 **/
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class RequestHeader {
    /**
     * Request header name
     */
    private String name;
    /**
     * Header description
     */
    private String desc;
    /**
     * Whether must
     */
    private boolean required = false;
}
